### PR TITLE
improve some example from opencypher tck

### PR DESCRIPTION
**Pull request by whitewum**

### CODE REVIEW

File: docs-2.0/3.ngql-guide/3.data-types/4.date-and-time.md [hunk 1]
```diff
@@ -59,3 +59,20 @@ Insert a vertex named "dut" with the foundation date now.
 ```ngql
 nebula> INSERT VERTEX school(name, found_time) VALUES "dut":("dut", now());
 ```
+
+```ngql
+nebula> WITH time({hour: 12, minute: 31, second: 14}) AS d RETURN d
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/4.date-and-time.md:64]: add `;` at the end of each query

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md [hunk 2]
```diff
@@ -4,6 +4,34 @@ You can set the properties for vertices or edges to `NULL`. Also, you can set `N
 
 If not specified, the property is set to `NULL` by default.
 
+## Logical operations with NULL
+
+<!-- The logcial operations with NULL is the same as OpenCypher. -->
+
+Here is the true table for AND, OR, XOR, and NOT.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/5.null.md:11]: truth table

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md [hunk 3]
```diff
@@ -4,6 +4,34 @@ You can set the properties for vertices or edges to `NULL`. Also, you can set `N
...
+
+| a | b | a AND b | a OR b | a XOR | NOT a |
+| - | - | - | - | - | - |
+| false|  null|  false | null | null | true |
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/5.null.md:15]: add a new line, a is false b is false

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md [hunk 4]
```diff
@@ -4,6 +4,34 @@ You can set the properties for vertices or edges to `NULL`. Also, you can set `N
...
+| null | false | false | null | null | null|
+| null | null | null | null | null | null|
+| null | true | null | true | null | null|
+
+## Comparisons with NULL
+
+TODO(incompatible)
+<!-- The comparison operations with NULL is incompatible with OpenCypher. -->
+
+## Operations and expression with NULL
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/5.null.md:29]: What about the `IN` operator and `null`?

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md [hunk 5]
```diff
@@ -4,6 +4,34 @@ You can set the properties for vertices or edges to `NULL`. Also, you can set `N
...
+
+TODO(incompatible)
+<!-- The  NULL operations and RETURN with NULL is incompatible with OpenCypher. -->
+
+
 ## Examples
 
 Create a tag named player. Specify the property name with `NOT NULL`. Skip the property age constraint.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/5.null.md:37]: Skip -> Ignore

File: docs-2.0/3.ngql-guide/3.data-types/6.list.md [hunk 6]
```diff
@@ -1,10 +1,10 @@
-# List
+# Lists
 
-List is a composite data type. A composite data type cannot be stored as properties. List is a sequence of values. Individual list elements can be accessed by their positions.
+List is a composite data type. List is a sequence of values. Individual list elements can be accessed by their positions.
 
-A list starts with a left square bracket `[` and ends with a right square bracket `]`. A list contains zero, one, or more expressions. List elements are separated from each other with a comma `,`. Whitespace around elements is ignored in list, thus line breaks, tab stops, and blanks can be used for formatting.
+> **openCypher Incompatible:** A composite data type (i.e. set, map, and list) cannot be stored as (vertex's or edge's) properties.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:5]: cannot>- CANNOT
as properties for vertices or edges.

File: docs-2.0/3.ngql-guide/3.data-types/6.list.md [hunk 7]
```diff
@@ -1,10 +1,10 @@
...
 
-> **CAUTION:** A composite data type (i.e. set, map, and list) cannot be stored as properties.
+A list starts with a left square bracket `[` and ends with a right square bracket `]`. A list contains zero, one, or more expressions. List elements are separated from each other with a comma `,`. Whitespace around elements is ignored in list, thus line breaks, tab stops, and blanks can be used for formatting.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:7]: with commas

File: docs-2.0/3.ngql-guide/3.data-types/8.map.md [hunk 8]
```diff
@@ -1,3 +1,21 @@
-# Map
+# Maps
 
 Map is a composite data type. A composite data type cannot be stored as properties. Maps are unordered collections of key-value pairs. In maps, the key is a string. The value can have any data type. You can get the map element by using `map['key']`.
+
+> **openCypher Incompatible:** A composite data type (i.e. map, and list) cannot be stored as (vertex's or edge's) properties.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/3.data-types/8.map.md:5]: ditto

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md [hunk 9]
```diff
@@ -1,26 +1,39 @@
-# Composite queries
+# Composite queries (clause structure)
 
-Composite queries put data from different queries together. They then use filters, group bys, or sorting before returning the combined return results. A composite query retrieves multiple levels of related information on existing queries and presents data as a single return result.
+Composite queries put data from different queries together. They then use filters, group-bys, or sorting before returning the combined return results. A composite query retrieves multiple levels of related information on existing queries and presents data as a single return result.
 
-Nebula Graph supports two ways to compose queries (or sub-queries):
+Nebula Graph supports three ways to compose queries (or sub-queries):
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md:5]: three methods.
My fault. 3q

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md [hunk 10]
```diff
@@ -1,26 +1,39 @@
...
+For example, a query composed of three sub-queries: `A B C`, `A | B | C` or ` A; B; C`. In that A is a read operation, B is a computation operation, and C is a write operation. If any part fails in the execution, the whole result is undefined. There is no rollback. What is written depends on the query executor.
+
+> Tips: openCypher has no requirement of `transaction`.
 
 ## Examples
 
+- openCypher style
+
+```ngql
+MATCH p=(v:player{name:"Tim Duncan"})--() \
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md:22]: nebula>

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md [hunk 11]
```diff
@@ -1,14 +1,31 @@
 # User-defined variables
 
-User-defined variables allows passing the result of one statement to another. User-defined variables are written as `$var_name`. The `var_name` consists of alphanumeric characters. Any other characters are not recommended currently.
+User-defined variables allows passing the result of one statement to another. User-defined variables are written as `$var_name`. The `var_name` consists of alphanumeric characters. Any other characters are not recommended.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md:3]: Any other characters are not recommended. -> Any other characters are not permitted.

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md [hunk 12]
```diff
@@ -1,14 +1,31 @@
...
+
+## openCypher variables
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md:5]: OpenCypher?

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md [hunk 13]
```diff
@@ -4,6 +4,34 @@ You can set the properties for vertices or edges to `NULL`. Also, you can set `N
...
+| a | b | a AND b | a OR b | a XOR | NOT a |
+| - | - | - | - | - | - |
+| false|  null|  false | null | null | true |
+| false | true | false | true | true | true|
+| true | false | false | true | true | false|
+| true|  null | null | true | null | false|
+| true | true | true | true | false | false|
+| null | false | false | null | null | null|
+| null | null | null | null | null | null|
+| null | true | null | true | null | null|
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/5.null.md:22]: Awesome!

File: docs-2.0/3.ngql-guide/3.data-types/6.list.md (hunk 6 above)
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:5]: >**OpenCypher compatibility:**

Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:5]: 's is not needed.

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md [hunk 14]
```diff
@@ -1,14 +1,31 @@
...
+```
+
+The user-defined variable in the preceding query is `v`.
+
+## nGQL extensions
 
 User-defined variables can only be used in one execution. For example, you can use user-defined variables in composite queries separated by semicolon `;` or pipe `|`. Details about composite queries, see [Composite queries](1.composite-queries.md).
 
-> **NOTE:** A user-defined variable is valid only at the current session and execution.
+> **Tips:** A user-defined variable is valid only at the current session and execution.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md:24]: Tips -> NOTE

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md [hunk 15]
```diff
@@ -82,4 +84,5 @@ nebula> GO FROM "player100" OVER follow YIELD follow._src, follow._dst, follow._
 +-------------+-------------+--------------+--------------+
 ```
 
-The preceding query returns all the neighbors of vertex `"player100"` over edge type `follow`, by referencing `follow._src` as the source vertex ID (which, of course, is `"player100"`) and `follow._dst` as the destination vertex ID.
+The preceding query returns all the neighbors of vertex `"player100"` over edge type `follow`,
+by referencing `follow._src` as the source vertex ID (which is `"player100"`) and `follow._dst` as the destination vertex ID.
```
Comment (Amber1990Zhang) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md:88]: why break the sentence?

File: docs-2.0/3.ngql-guide/3.data-types/6.list.md [hunk 16]
```diff
@@ -1,10 +1,10 @@
-# List
+# Lists
 
-List is a composite data type. A composite data type cannot be stored as properties. List is a sequence of values. Individual list elements can be accessed by their positions.
+List is a composite data type. List is a sequence of values. Individual list elements can be accessed by their positions.
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:3]: The list is a composite data type. A list is a sequence of values.

File: docs-2.0/3.ngql-guide/3.data-types/6.list.md [hunk 17]
```diff
@@ -95,6 +95,10 @@ nebula> YIELD size([1,2,3]);
 
 ## openCypher compatibility
 
+TODO(incompatible)
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/6.list.md:98]: mark

File: docs-2.0/3.ngql-guide/3.data-types/7.set.md [hunk 18]
```diff
@@ -1,5 +1,11 @@
+TODO(notimplemented)
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/3.data-types/7.set.md:1]: mark

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md [hunk 19]
```diff
@@ -1,26 +1,39 @@
...
+## Composite queries are not `transactional` queries (in SQL/Cypher).
 
-For example, a query composed of three sub-queries: A | B | C. In that A is a read operation, B is a computation operation, and C is a write operation. If any part fails in the execution, the whole result can be undefined. Currently, there is no rollback. What is written depends on the query executor.
+For example, a query composed of three sub-queries: `A B C`, `A | B | C` or ` A; B; C`. In that A is a read operation, B is a computation operation, and C is a write operation. If any part fails in the execution, the whole result is undefined. There is no rollback. What is written depends on the query executor.
+
+> Tips: openCypher has no requirement of `transaction`.
 
 ## Examples
 
+- openCypher style
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/1.composite-queries.md:19]: OpenCypher when it is leading a sentence or a title

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md [hunk 20]
```diff
@@ -1,14 +1,31 @@
 # User-defined variables
 
-User-defined variables allows passing the result of one statement to another. User-defined variables are written as `$var_name`. The `var_name` consists of alphanumeric characters. Any other characters are not recommended currently.
+User-defined variables allows passing the result of one statement to another. User-defined variables are written as `$var_name`. The `var_name` consists of alphanumeric characters. Any other characters are not recommended.
+
+## openCypher variables
+
+In openCypher, when you refer to a pattern, you need to name it first. The name you give to the pattern is a variable. For example:
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md:7]: >you need to name it first

Why? I don't think we need to name a pattern. `v` represents a vertex, not the whole pattern.

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md [hunk 21]
```diff
@@ -1,14 +1,31 @@
...
 
 User-defined variables can only be used in one execution. For example, you can use user-defined variables in composite queries separated by semicolon `;` or pipe `|`. Details about composite queries, see [Composite queries](1.composite-queries.md).
 
-> **NOTE:** A user-defined variable is valid only at the current session and execution.
+> **Tips:** A user-defined variable is valid only at the current session and execution.
 
 A user-defined variable in one statement **CANNOT** be used in either other clients or other executions. The statement that defines the user-defined variable and the statement that uses it must be submitted together. When this session ends, the user-defined variable is automatically expired.
 
-User-defined variables are case-sensitive.
+User-defined variables are **case-sensitive**.
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/2.user-defined-variables.md:28]: Why bold?

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md [hunk 22]
```diff
@@ -82,4 +84,5 @@ nebula> GO FROM "player100" OVER follow YIELD follow._src, follow._dst, follow._
 +-------------+-------------+--------------+--------------+
 ```
 
-The preceding query returns all the neighbors of vertex `"player100"` over edge type `follow`, by referencing `follow._src` as the source vertex ID (which, of course, is `"player100"`) and `follow._dst` as the destination vertex ID.
+The preceding query returns all the neighbors of vertex `"player100"` over edge type `follow`,
```
Comment (randomJoe211) [inline docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md:87]: over the `follow` edges?

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md (hunk 15 above)
Comment (randomJoe211) [reply docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md:88]: The mysterious break.

**Review by randomJoe211 — COMMENTED** [review]
👍

File: docs-2.0/3.ngql-guide/3.data-types/5.null.md (hunk 4 above)
Comment (whitewum) [reply docs-2.0/3.ngql-guide/3.data-types/5.null.md:29]: Current (2021.2.1) behavior is different with openCypher. 
I leave a TODO(coding) in the doc.

File: docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md (hunk 15 above)
Comment (whitewum) [reply docs-2.0/3.ngql-guide/4.variable-and-composite-queries/3.property-reference.md:88]: It's too long. One line break is the same as a blackspace.

**Review by randomJoe211 — APPROVED** [review]

**Review by Amber1990Zhang — APPROVED** [review]
LGTM

**Pull request merged**